plan/do-not-destroy-s3.rego

```diff
@@ -0,0 +1,28 @@
+package spacelift
+
+# This policy is a plan policy, it will validate the resources during the plan phase.
+# More details at: https://docs.spacelift.io/concepts/policy/terraform-plan-policy
+# The "deny" rule fires when a specified resource is being deleted.
+# The result is a formatted message with the address of the offending resource.
+
+deny[sprintf(message, [resource.address])] { 
+  # Define the error message format
+  message := "do not delete %s" 
+
+  # Loop over each resource change in the plan
+  resource := input.terraform.resource_changes[_]
+
+  # Define a set of resource types for which deletions should be prevented.
```

Comment:
This policy was designed to be extensible, which is excellent, but then I would name it accordingly ("Do not delete stateful resources"?) and include more stateful resources such as AWS RDS instances, AWS EFS volumes, AWS DynamoDB tables, etc.

Reply:
changed to include these resources